src/app/machines/views/MachineList/MachineListControls/MachinesFilterAccordion/MachinesFilterOptions/MachinesFilterOptions.tsx

```diff
@@ -0,0 +1,112 @@
+import type { ReactNode } from "react";
+import { useEffect } from "react";
+
+import { Button, List, Spinner } from "@canonical/react-components";
+import classNames from "classnames";
+import { useDispatch, useSelector } from "react-redux";
+
+import { actions as machineActions } from "app/store/machine";
+import machineSelectors from "app/store/machine/selectors";
+import type {
+  FilterGroup,
+  FilterGroupOptionType,
+} from "app/store/machine/types";
+import { FilterMachines } from "app/store/machine/utils";
+import type { RootState } from "app/store/root/types";
+import { simpleSortByKey } from "app/utils";
+
+export enum Label {
+  False = "false",
+  Loading = "Loading...",
+  None = "No options",
+  True = "true",
+}
+
+type Props = {
+  group: FilterGroup["key"];
+  searchText?: string;
+  setSearchText: (searchText: string) => void;
+};
+
+const MachinesFilterOptions = ({
+  group,
+  searchText,
+  setSearchText,
+}: Props): JSX.Element => {
+  const dispatch = useDispatch();
+  const filterOptions = useSelector((state: RootState) =>
+    machineSelectors.filterOptions(state, group)
+  );
+  const optionsLoading = useSelector((state: RootState) =>
+    machineSelectors.filterOptionsLoading(state, group)
+  );
+  const optionsLoaded = useSelector((state: RootState) =>
+    machineSelectors.filterOptionsLoaded(state, group)
+  );
+  const currentFilters = FilterMachines.getCurrentFilters(searchText);
+
+  useEffect(() => {
+    if (!optionsLoading && !optionsLoaded) {
+      dispatch(machineActions.filterOptions(group));
+    } else {
+    }
+  }, [dispatch, group, optionsLoading, optionsLoaded]);
+
+  let items: ReactNode[];
+  if (optionsLoading) {
+    items = [
+      <span className="filter-accordion__item u-align-text--center">
+        <Spinner text={Label.Loading} />
+      </span>,
+    ];
+  } else if (!filterOptions || filterOptions.length === 0) {
+    items = [
+      <span className="filter-accordion__item u-text--muted">
+        {Label.None}
+      </span>,
+    ];
+  } else {
+    items = [...filterOptions].sort(simpleSortByKey("key")).map((filter) => {
```

Comment:
I might be missing something, but is there anything that stops us from returning sorted filters by default from the filterOptions selector?

Reply:
No nothing at all, that's a good idea. I'll update it.